### PR TITLE
Handle lists of available tsos and business processes

### PR DESCRIPTION
**Pull request by FranckLecuyer**
https://dev.azure.com/GridSuite/MVP/_sprints/backlog/MVP%20Team/MVP/Sprint%2027?workitem=460

### CODE REVIEW

**Review by ChamseddineBhd — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java [hunk 1]
```diff
@@ -95,4 +111,48 @@ String importBoundary(MultipartFile mpfFile) {
...
+            tsosRepository.insert(entity);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    void importBusinessProcesses(MultipartFile businessProcessesFile) {
+        try {
+            ByteBuffer buf = ByteBuffer.wrap(businessProcessesFile.getBytes());
+            BusinessProcessesListEntity entity = new BusinessProcessesListEntity(BUSINESS_PROCESS_LIST_NAME, buf);
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryService.java:152]: Yes

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java [hunk 2]
```diff
@@ -89,4 +90,40 @@
...
         return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(cgmesBoundaryService.boundaryExists(boundaryId));
     }
+
+    @GetMapping(value = "/tsos", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Get list of all available tsos", response = String.class)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of all available tsos")})
+    public ResponseEntity<Set<String>> getTsos() {
+        Optional<Set<String>> tsos = cgmesBoundaryService.getTsos();
+        if (!tsos.isPresent()) {
+            throw new PowsyblException("No available list of tsos");
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java]: empty set is now returned

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java [hunk 3]
```diff
@@ -89,4 +90,40 @@
...
+        }
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(tsos.get());
+    }
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java:101]: Done

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java [hunk 4]
```diff
@@ -89,4 +90,40 @@
...
+    @GetMapping(value = "/business-processes", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Get list of all available business processes", response = String.class)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of all available business processes")})
+    public ResponseEntity<Set<String>> getBusinessProcesses() {
+        Optional<Set<String>> businessProcesses = cgmesBoundaryService.getBusinessProcesses();
+        if (!businessProcesses.isPresent()) {
+            throw new PowsyblException("No available list of business processes");
+        }
+        return ResponseEntity.ok().contentType(MediaType.APPLICATION_JSON).body(businessProcesses.get());
+    }
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java:116]: Done

File: src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java [hunk 5]
```diff
@@ -89,4 +90,40 @@
...
+        return ResponseEntity.ok().build();
+    }
+
+    @GetMapping(value = "/business-processes", produces = MediaType.APPLICATION_JSON_VALUE)
+    @ApiOperation(value = "Get list of all available business processes", response = String.class)
+    @ApiResponses(value = {@ApiResponse(code = 200, message = "The list of all available business processes")})
+    public ResponseEntity<Set<String>> getBusinessProcesses() {
+        Optional<Set<String>> businessProcesses = cgmesBoundaryService.getBusinessProcesses();
+        if (!businessProcesses.isPresent()) {
+            throw new PowsyblException("No available list of business processes");
```
[the remark this answers is not in the document]
Comment (FranckLecuyer) [reply src/main/java/org/gridsuite/cgmes/boundary/server/CgmesBoundaryController.java]: Done

**Review by JacquesBorsenberger — APPROVED** [review]

**Review by ChamseddineBhd — APPROVED** [review]

**Pull request merged**